### ISSUE TITLE
CSRF security check with secure urls reloads error page in loop [BUG]

### ISSUE THREAD

**Issue by aladin-29** [roundcube/roundcubemail#8617]
Hello,
we are having a strange issue with secure urls check in roundcube. 
I was wondering if any of you have faced similiar issue.

So, in the config we have set:
`$config['use_secure_urls'] = true;`

and in the ngnix.conf 
` location / {
                rewrite "^/[a-zA-Z0-9]{16}/(.*)" /$1 break;
}`

and it works! 
But when we remove manually the random Token from the url, it just continue to reload in loop.
so, from https://mysite.com/abcdefgh12345678/ to -> https://mysite.com/ 
it does show the error page `[Click here to resume your previous session.] ` 
but for few ms and starts reload it in loop.

If it can help, I've checked the line where this function is called `$RCMAIL->request_security_check();`

https://github.com/roundcube/roundcubemail/blob/49cf2286d2c87eab0875819a63da463a590b7039/index.php#L269

which call raise_error function
https://github.com/roundcube/roundcubemail/blob/49cf2286d2c87eab0875819a63da463a590b7039/program/include/rcmail.php#L1217

if i comment that line, it stops to reload in loop but this way i guess the check is not carried out, infact it shows the normal page of our application.

Any kind of suggestion will be helpful
Thanks and Best Regards

P.S. i've tried also disabling all our plugins but the results are the same.

**EDIT:**
I've done some further check. so we have this function which check csrf prevention
https://github.com/roundcube/roundcubemail/blob/49cf2286d2c87eab0875819a63da463a590b7039/program/include/rcmail.php#L1212

```
    public function request_security_check($mode = rcube_utils::INPUT_POST)
    {
        // check request token
        if (!$this->check_request($mode)) {
            $error = array('code' => 403, 'message' => "Request security check failed");
            self::raise_error($error, false, true);
        }
    }
```
which throw correctly 403 error but the problem is how it is handled by this one:

https://github.com/roundcube/roundcubemail/blob/49cf2286d2c87eab0875819a63da463a590b7039/program/js/app.js#L9355

```
 // 403 Forbidden response (CSRF prevention) - reload the page.
    // In case there's a new valid session it will be used, otherwise
    // login form will be presented (#1488960).
    if (request.status == 403) {
      (this.is_framed() ? parent : window).location.reload();
      return;
    }
```
i've put some console log and i've seen that both parent and window location just reload the current url (https://mysite.com/ , which is wrong in this case) and as a result it goes in loop. 

So, to summarise:
--> from https://mysite.com/abcdefgh12345678/ to -> https://mysite.com/ 
--> it throws 403 and reload again https://mysite.com/  
--> which throw again 403 and continue to reload the page without the token creating the infinite loop 

In my opinion, it should only show the error page and stop, so the user has the possibility to click on the link `[Click here to resume your previous session.]`

It should be considered as a bug , i guess.
```
 | Roundcube Webmail IMAP Client                                          
 | Version 1.4.11
```
Please let me know, if you need further details,
Thanks and Best Regards.

**Comment by alecpl** [comment]
I'd suggest to try with 1.6.0. 1.4 is in low maintenance mode.
